lib/ansible/module_utils/mysql.py

```diff
@@ -36,13 +37,29 @@
     mysqldb_found = False
 
 
+def _parse_from_mysql_config_file(cnf):
+    if not cnf:
+        return None
+
+    cp = configparser.ConfigParser()
+    cp.read(cnf)
+
+    return cp
+
+
 def mysql_connect(module, login_user=None, login_password=None, config_file='', ssl_cert=None, ssl_key=None, ssl_ca=None, db=None, cursor_class=None,
                   connect_timeout=30):
     config = {}
 
     if ssl_ca is not None or ssl_key is not None or ssl_cert is not None:
         config['ssl'] = {}
 
+    cp = _parse_from_mysql_config_file(config_file)
+    if cp and cp.has_section('client'):
+        module.params['login_unix_socket'] = cp.get('client', 'socket', fallback=module.params['login_unix_socket'])
+        module.params['login_host'] = cp.get('client', 'host', fallback=module.params['login_host'])
+        module.params['login_port'] = cp.getint('client', 'port', fallback=module.params['login_port'])
```

Comment:
This looks like it is unfortunately not quite this simple.  We want the following fallbacks:
* If the user specifies a value explicitly in the playbook, use that
* If the config file has it, then use that.
* If both of those do not have a value, then use the module's defaults.

This likely needs changes both here and in the mysql modules themselves.  The change here will look something like this (repeat for each of the values we're setting):
``` python
if cp an cp.has_section('client'):
    if module.params['login_host'] is None:
        module.params['login_host'] = cp.getint('client', 'host', fallback='localhost')
    [....]
```

Inside of the mysql modules we'll have to change teh module's argument_spec.  For instance, mysql_db.py has this:
``` python
        argument_spec=dict(
            login_user=dict(default=None),
            login_password=dict(default=None, no_log=True),
            login_host=dict(default="localhost"),
            [...]
```

We would need to change it to this:
``` python
        argument_spec=dict(
            login_user=dict(default=None),
            login_password=dict(default=None, no_log=True),
            login_host=dict(default=None),
            [...]
```

Reply:
@abadger Okay! I remove `login_host` default value and set 'socket', 'host' when these value is `None`. Thanks :)